chowpronto_frontend/src/patron/pages/confirmOrder/index.js

```diff
@@ -1,7 +1,108 @@
+import Login from "./components/Login";
+import UserDetailsForm from "../../components/UserDetailsForm";
+import useAuth from "../../../hooks/useAuth";
+import { MenuContext } from "../../../state/MenuContext";
 
 const ConfirmOrderPage = (props) => {
-  return <h1>Confirm Order Page</h1>;
+  const { getUser, login, logout } = useAuth();
+  const user = false;
+  const { state, dispatch } = useContext(MenuContext);
+  const [userData, setUserData] = useState({});
+  function saveOrder(returnedData) {
+    console.log("returnedData arguments to function", returnedData);
+    fetch("/api/orders/order", {
+      method: "POST",
+      headers: {
+        Authorization: `Bearer ${userData.token}`,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        cart: [
+          { menuItemId: "5ec0180b61361e716d8bf4a2", quantity: 5 },
+          { menuItemId: "5ec0180b61361e716d8bf4ba", quantity: 2 },
+          { menuItemId: "5ec0180b61361e716d8bf4cf", quantity: 3 },
+        ],
+        deliveryDetails: {
+          email: "kosenchiha@gmail.com",
+          name: "hjkhkg",
+          phone: "24255222244",
+          address: "1/2 flat, 20 Hope Street, Glasqow, UK",
+          postcode: "hjkhkg",
+        },
+        patronId: returnedData.patron._id,
+      }),
+    })
+      .then((res) => res.json())
+      .then((data) => console.log("returned from createOrder", data));
+  }
+  function handleSubmit(e) {
+    e.preventDefault();
+    if (!user) {
+      fetch("/api/patrons/signup", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          email: "def@gmail.com",
+          name: "New Patron",
+          password: "12345678",
+          phone: "+12-3457-8910",
+          address: "123 Flat, 12 Hope Street, Faith City, Wanderland",
+          postcode: "W 765 HS",
+          role: "GUEST",
+        }),
+      })
+        .then((res) => res.json())
+        .then(async (data) => {
+          console.log("data from async", data);
+          await setUserData(data);
+          saveOrder(data);
+          //return data;
+        })
+        // .then((returnedData) => {
+        //   saveOrder(returnedData);
+        // })
+        .catch((err) => console.log("err", err));
+    } else {
+      try {
+        saveOrder();
+      } catch (err) {
+        console.log("err", err);
+      }
+    }
+    // at this point the user should be stored on state
+
+    //  headers: {
+    //       Authorization: `Bearer ${token}`,
+    //       "Content-Type": "application/json"
+    //     }
+  }
+  return (
+    <PageLayout>
+      <StyledSidebar style={{ flex: 1.5 }}>
+        <OrderDetails />
+        <Login />
+      </StyledSidebar>
+      <StyledPageMain>
+        <UserDetailsForm />
+        <SubmitButton title="submit" primary onClick={(e) => handleSubmit(e)} />
+      </StyledPageMain>
+      <input type="password" />
+    </PageLayout>
```

Comment:
The **input type="password"** is currently spanning the whole component.

Reply:
I will go through & sort out the form layout in a later issue, maybe with some help from yourself. It looks like I may be swapping out some components fro yours anyway but this pull is primarily concerned with getting the logic working, a secondary password input needs to be added and these could be placed side by side.

Reply:
Agreed, we can clean this up later.  

Let's merge this PR once you're happy with the logic and context. Great job!